pkg/scheduler/scheduler.go

```diff
@@ -309,8 +309,20 @@ func filterWorkersByResources(workers []*types.Worker, request *types.ContainerR
 
 		if requiresGPU {
 			// Validate GPU resource availability
+
 			priorityModifier, validGpu := gpuRequestsMap[worker.Gpu]
-			if !validGpu || worker.FreeGpuCount < request.GpuCount {
+			if !validGpu {
```

Comment:
can't we simplify this by saying if requiresGpu && isGpuWorker, we filter it in?